ui/src/app/subnets/views/FabricDetails/FabricVLANs/FabricVLANs.tsx

```diff
@@ -1,7 +1,149 @@
-import TitledSection from "app/base/components/TitledSection";
+import { useEffect } from "react";
 
-const FabricVLANs = (): JSX.Element => (
-  <TitledSection title="VLANs on this fabric" />
-);
+import { Strip, MainTable } from "@canonical/react-components";
+import { useDispatch, useSelector } from "react-redux";
+
+import SpaceLink from "app/base/components/SpaceLink";
+import SubnetLink from "app/base/components/SubnetLink";
+import VLANLink from "app/base/components/VLANLink";
+import { useId } from "app/base/hooks/base";
+import type { Fabric } from "app/store/fabric/types";
+import { actions as spaceActions } from "app/store/space";
+import spaceSelectors from "app/store/space/selectors";
+import type { Space } from "app/store/space/types";
+import { actions as subnetActions } from "app/store/subnet";
+import subnetSelectors from "app/store/subnet/selectors";
+import type { Subnet } from "app/store/subnet/types";
+import { actions as vlanActions } from "app/store/vlan";
+import vlanSelectors from "app/store/vlan/selectors";
+import type { VLAN } from "app/store/vlan/types";
+
+type Columns = {
+  available: string | null;
+  space: Space | undefined;
+  subnet: Subnet | null;
+  vlan: VLAN;
+};
+
+const getAvailableIPs = (subnet: Subnet | null | undefined): string => {
+  if (!subnet) {
+    return "Unconfigured";
+  } else {
+    return `${subnet?.statistics?.available_string}`;
+  }
+};
+
+const getSpaceById = (
+  spaces: Space[],
+  spaceId: Space["id"] | null
+): Space | undefined => {
+  return spaces.find((space) => space?.id === spaceId);
+};
+
+const getSubnetsInVLAN = (subnets: Subnet[], vlanId: VLAN["id"]): Subnet[] =>
+  subnets.filter((subnet) => subnet.vlan === vlanId);
+
+const getVLANsInFabric = (vlans: VLAN[], fabricId: Fabric["id"]): VLAN[] =>
+  vlans.filter((vlan) => vlan.fabric === fabricId);
```

Comment:
You should be able to import most of the util functions you added to this file, they already exist, e.g. https://github.com/canonical-web-and-design/maas-ui/blob/9d880a47c3d8c5b980b10fca8862e3771141d982/ui/src/app/store/subnet/utils.ts#L51
Unless there's a reason why you duplicated them in this file?